core/blockchain.go

```diff
@@ -1679,6 +1689,12 @@ func (bc *BlockChain) writeBlockWithState(block *types.Block, receipts []*types.
 		diffLayer.Receipts = receipts
 		diffLayer.BlockHash = block.Hash()
 		diffLayer.Number = block.NumberU64()
+
+		sort.Sort(types.DiffCodeSlice(diffLayer.Codes))
+		sort.Sort(common.AddressSlice(diffLayer.Destructs))
+		sort.Sort(types.DiffAccountSlice(diffLayer.Accounts))
+		sort.Sort(types.DiffStorageSlice(diffLayer.Storages))
+
```

Comment:
Since `sort` is a heavy operation, can we move this to another routine, so that it doesn't have to block the processing of blockchain?

Reply:
Sure, I will move these sort functions into cacheDiffLayer function, and add a sorted flag, if need sort, create a go routine to call cacheDiffLayer, otherwise call directly.

Reply:
updated